### ISSUE TITLE
One app handle other app's push notification

### ISSUE THREAD

**Issue by vishaljadav24** [evollu/react-native-fcm#640]
**# Step to produce issue:**
- I am working with two apps. For this, I create two Firebase project and create firebase app respectively.
- Both apps are installed on my phone.
- Server key is different by project but Fcm token is same.
- Server sends a notification to my one app by Fcm token and Server key. 
- If my second app is open then below code of my second app execute
```js
FCM.on(FCMEvent.Notification, (notif) => {
// this code execute
}
```
- I don't know this issue is by server-side or app side. So please guide me, how to solve this issue?

**Comment by evollu** [comment]
are you sure you are using the right server key?
You shouldn't get the same fcm token for different app.

**Comment by vishaljadav24** [comment]
Yes. I am sure for server key.

**Comment by evollu** [comment]
Server key is different by project but Fcm token is same. can you verify it? your 2 apps have different app id right?

**Comment by vishaljadav24** [comment]
Yes @evollu , I verify it and app id also different. I searched a lot for this issue. But nothing found which solved my issue. So please help me.

**Comment by evollu** [comment]
try delete and reinstall both apps and see if you get the same fcm token. They should be different

**Comment by vishaljadav24** [comment]
I got different Fcm tokens after uninstall app but issue not solve.

**Comment by vishaljadav24** [comment]
We made some changes in code as per below and it worked!  For Title and Text send those data under `notification` Key. 
```js
{
	"to": "fcmToken",
	"priority": "high",
	"data": {
		"type": "BOOK",
		"delivery_id": 71,
		"start_time": 1510817811,
		"end_time": 1510817840,
                "sound": "default",
                "title": "New Delivery Booked!",
                "text": "New delivery is booked accept before its too late!"
	}
}
```
changable code
```js
{
	"to": "fcmToken",
	"priority": "high",
	"data": {
		"type": "BOOK",
		"delivery_id": 71,
		"start_time": 1510817811,
		"end_time": 1510817840,
                "sound": "default"
	},
	"notification": {
	"title": "New Delivery Booked!",
         "text": "New delivery is booked accept before its too late!"		
	}
}

**Comment by evollu** [comment]
cool... I don't know the real cause though...perhaps some issue with firebase

**Comment by vishaljadav24** [comment]
I face this issue again. The issue comes when one app is running in background and second app is in the foreground. For both the app push is coming. Attaching the screenshot for better understanding.
If I kill the second app and only one app is in active state then issue is not coming.
![fcm_push_issue](https://user-images.githubusercontent.com/24474146/33429182-c198642c-d5f1-11e7-9994-e11f31ee12a7.png)

**Comment by evollu** [comment]
ok. I might know what is going on now. Broadcast event seems to be sent to all applications so 
1. if one of your app received refresh token event, the other app receives too. this might explain why you have same firebase token for 2 apps. Can you verify that?

**Comment by evollu** [comment]
@vishaljadav24 try master branch
